modules/database/test/std/rec/aiTest.c

```diff
@@ -0,0 +1,429 @@
+    testdbPutFieldOk("test_ai_rec.ROFF", DBF_SHORT, roff);
+    testdbPutFieldOk("test_ai_rec.ASLO", DBF_SHORT, aslo);
+    testdbPutFieldOk("test_ai_rec.AOFF", DBF_SHORT, aoff);
+    
+    /* verify conversion */
+    testdbPutFieldOk("test_ai_link_rec.VAL", DBF_SHORT, rval);
+    testdbGetFieldEqual("test_ai_rec.VAL", DBF_SHORT, val);
+    testdbGetFieldEqual("test_ai_rec.RVAL", DBF_SHORT, rval);
+
+    // number of tests = 9
+}
+
+static void test_slope_linr_unit_conversion(void){
+    const short roff = 1;
+    const short aslo = 3;
+    const short aoff = 99;
+    const short eslo = 3;
+    const short eoff = 2;
+
+    const short rval = 7;
+    const short val = ((((rval + roff) * aslo) + aoff) * eslo) + eoff;
+
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_ai_rec.INP", DBF_STRING, "test_ai_link_rec.VAL");
+    testdbPutFieldOk("test_ai_link_rec.FLNK", DBF_STRING, "test_ai_rec");
+
+    /* set unit conversion parameters */
+    testdbPutFieldOk("test_ai_rec.ROFF", DBF_LONG, roff);
+    testdbPutFieldOk("test_ai_rec.ASLO", DBF_LONG, aslo);
+    testdbPutFieldOk("test_ai_rec.AOFF", DBF_LONG, aoff);
+    testdbPutFieldOk("test_ai_rec.ESLO", DBF_LONG, eslo);
+    testdbPutFieldOk("test_ai_rec.EOFF", DBF_LONG, eoff);
+    testdbPutFieldOk("test_ai_rec.LINR", DBF_LONG, menuConvertSLOPE);
+    
+    /* verify conversion */
+    testdbPutFieldOk("test_ai_link_rec.VAL", DBF_LONG, rval);
+    testdbGetFieldEqual("test_ai_rec.VAL", DBF_LONG, val);
+    testdbGetFieldEqual("test_ai_rec.RVAL", DBF_LONG, rval);
+
+    // number of tests = 12
+}
+
+static void test_linear_linr_unit_conversion(void){
+    const long roff = 6;
+    const long aslo = 2;
+    const long aoff = 19;
+    const long eslo = 4;
+    const long eoff = 1;
+    const long egul = 9999;
+    const long eguf = -1000;
+
+    const long rval = 2;
+    /* Since our raw soft input does not support EGUL and EGUF conversion LINEAR should work like SLOPE */
+    const long val = ((((rval + roff) * aslo) + aoff) * eslo) + eoff;
+
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_ai_rec.INP", DBF_STRING, "test_ai_link_rec.VAL");
+    testdbPutFieldOk("test_ai_link_rec.FLNK", DBF_STRING, "test_ai_rec");
+
+    /* set unit conversion parameters */
+    testdbPutFieldOk("test_ai_rec.ROFF", DBF_LONG, roff);
+    testdbPutFieldOk("test_ai_rec.ASLO", DBF_LONG, aslo);
+    testdbPutFieldOk("test_ai_rec.AOFF", DBF_LONG, aoff);
+    testdbPutFieldOk("test_ai_rec.ESLO", DBF_LONG, eslo);
+    testdbPutFieldOk("test_ai_rec.EOFF", DBF_LONG, eoff);
+    /* Since our raw soft input does not support EGUL and EGUF conversion we set them to lagre values here just to check they dont break anything */
+    testdbPutFieldOk("test_ai_rec.EGUL", DBF_LONG, egul);
+    testdbPutFieldOk("test_ai_rec.EGUF", DBF_LONG, eguf);
```

Comment:
A suggestion here could be to "mock" a device support calculation by using calc records to perform the EGUL/EGUF calculation.

Reply:
I am not convinced this would add any value to the test code. The calc record would just set the same conversion fields I am already testing.